.github/workflows/setup_venvs.yaml

```diff
@@ -0,0 +1,75 @@
+name: Setup virtual environments
+on: 
+  workflow_call:
+    inputs:
+      os:
+        required: true
+        type: string
+        default: ubuntu-latest
+  # workflow_dispatch:
+  # workflow_run:
+  #   workflows:
+  #     - Setup basic software environment
+  #   types:
+  #     - completed
+  # push:
+  # pull_request:
+  #   types: [review_requested, ready_for_review]
+
+jobs:
+
+  Setup_software_basis_py310_latest:
+    uses: ./.github/workflows/setup_basic_software.yaml
+    with:
+      os: ubuntu-latest
+      
+      # with:
+      #   cache-name: "cache-software-basis"
+
+  venv_matrix:
+    runs-on: ubuntu-latest
+    needs: Setup_software_basis_py310_latest
+    environment: Software Build
+    strategy:
+      matrix:
+        include:
+          - venv: "columnar"
+            name: "columnar"
+            hashFiles: |
+              **/sandboxes/venv_columnar*.sh
+              **/sandboxes/columnar.txt
+              **/sandboxes/_setup_venv.sh
+          # - venv: "docs"
+          #   name: "docs"
+          #   hashFiles: |
+          #     **/sandboxes/venv_docs*.sh
+          #     **/requirements_prod.txt
+          #     **/requirements_dev.txt
+          #     **/sandboxes/columnar.txt
+          #     **/sandboxes/ml_tf.txt
+          #     **/sandboxes/_setup_venv.sh
```

Comment:
This probably need to be adapted, but in principle we also want to include the `docs` venv in the matrix, right?